### DISCUSSION TITLE
Menu 组件的 itemSelectedColor, itemSelectedBg 会影响上级菜单的字体颜色

### DISCUSSION THREAD

**Discussion by ekuku1984** [ant-design/ant-design#52094] [General]
根据 UI 的设计， 需要把选定项的背景设置为主题色， 字体色设置为白色， 通过 Menu 的 itemSelectedColor, itemSelectedBg 两个 token 很容易实现， 但是在测试的时候， 发现选定子级的时候， 上级菜单的字体颜色变成白色了：

![image](https://github.com/user-attachments/assets/98df4bf7-029a-4d1e-b235-28118ea71ee9)

![image](https://github.com/user-attachments/assets/7a07e0e8-a66c-415d-b6a3-6310b9af2fa0)

这个不知道是特意设计成这样还是出于什么考虑， 现在不知道要怎么解决这个问题

**Comment by petercat-assistant** [comment]
您提到的问题是关于使用 Ant Design 的 `Menu` 组件时，设置 `itemSelectedColor` 和 `itemSelectedBg` 后，选定子级菜单时上级菜单的字体颜色也变成白色的问题。这可能是由于样式的继承或特定的样式规则导致的。

要解决这个问题，您可以尝试以下几种方法：

1. **自定义样式覆盖**：
   - 您可以通过自定义 CSS 来覆盖默认的样式行为。为上级菜单项设置特定的字体颜色，以确保它们在子菜单项被选中时不会受到影响。

2. **检查样式优先级**：
   - 确保您的自定义样式具有足够的优先级，以覆盖 Ant Design 的默认样式。可以通过增加 CSS 选择器的权重或使用 `!important` 来实现。

3. **使用 Ant Design 的主题功能**：
   - 如果您使用的是 Ant Design 的主题功能，可以尝试调整主题配置，以确保选定状态的样式不会影响到上级菜单。

如果这些方法无法解决您的问题，可能需要查看具体的代码实现或样式配置，以便更准确地定位问题。

如果您有进一步的代码示例或配置细节，可以提供给我，我将更好地帮助您解决问题。

<details>
<summary>🪧 Tips</summary>
For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.
</details>
